### PR TITLE
CSHARP-3044: Remove compile task from evergreen configuration

### PR DESCRIPTION
**Pull request by vincentkam**
https://jira.mongodb.org/browse/CSHARP-3044
https://evergreen.mongodb.com/version/5ea06ebca4cf4779f445d7a5

### CODE REVIEW

**Comment by vincentkam** [comment]
Ready for review 🐈

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -596,9 +596,6 @@ tasks:
         # - func: upload-build
 
     - name: test
```
Comment (MikalaiMazurenka) [inline evergreen/evergreen.yml:598]: Should we also remove dependency from [plain-auth-tests](https://github.com/vincentkam/mongo-csharp-driver/blob/50b77640f3a2c729f4f6306a7cd473e6ea824835/evergreen/evergreen.yml#L610), [aws-auth-tests](https://github.com/vincentkam/mongo-csharp-driver/blob/50b77640f3a2c729f4f6306a7cd473e6ea824835/evergreen/evergreen.yml#L617) and [atlas-connectivity-tests](https://github.com/vincentkam/mongo-csharp-driver/blob/50b77640f3a2c729f4f6306a7cd473e6ea824835/evergreen/evergreen.yml#L603)?

**Review by MikalaiMazurenka — APPROVED** [review]
Minor suggestion, feel free to ignore it.
LGTM!

File: evergreen/evergreen.yml (hunk 1 above)
Comment (vincentkam) [reply evergreen/evergreen.yml:598]: Ah, a reasonable idea! Thanks! I've taken your suggestion :)

**Review by DmitryLukyanov — APPROVED** [review]
Lgtm

**Pull request merged**